### ISSUE TITLE
secretmanager: List all secrets API takes long time to return

### ISSUE THREAD

**Issue by pravinpushkar** [googleapis/google-cloud-go#6982]
**Client**

Secretemanager's go SDK 
API -  ListSecrets

**Description**
I have close to 40 secrets in my secret store and I am using golang sdk's `ListSecrets` to fetch all secrets along with their value.

This takes somewhere **20-25 secs** to return all secrets. Here is my code - 
https://github.com/pravinpushkar/components-contrib/blob/ad8f21d3b246c2bd11addf34c39687a9bc816104/secretstores/gcp/secretmanager/secretmanager.go#L132

which is also similar to this - https://github.com/GoogleCloudPlatform/golang-samples/blob/692f934095411b2cc5d4a8ab0c8cc804034c31f6/secretmanager/list_secrets.go

Is their recommended or better way to handle this operation ?

I am using secretemanager#1.7.0

**Comment by pravinpushkar** [comment]
Hello, is there any thoughts on this ?

**Comment by codyoss** [comment]
I modified your code slightly and was unable to reproduce this. I would suggest adding some more telemetry to your code to try to figure out where the time is being spent. I was able to list and pull the secret versions for about 50 secrets in about 3-5 seconds, which seems reasonable. If you have high network latency being the calling code and google cloud I could see the number being higher though. You are both listing all these secrets and pulling version data for all of them so you are making around 40 reqs. If you are still having issues after adding in more debugging logic I would suggest reaching out through a support contract if you have one. I don't think there is anything in the library code that is causing this latency though. Hope that helps get you going in the right direction.
